### PR TITLE
Isaev Dmitriy Lab 2 Var 2

### PR DESCRIPTION
**Pull request by 4isper**

### CODE REVIEW

File: llvm/labs/lab2/isaev_dmitriy/inline_func.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        CurrentBB =
+            llvm::BasicBlock::Create(Func.getContext(), "", &Func, PostCallBB);
+        ValueMap[&CalleeBB] = CurrentBB;
+
+        Builder.SetInsertPoint(CurrentBB);
+        for (llvm::Instruction &Inst : CalleeBB) {
+          if (!Inst.isTerminator()) {
+            llvm::Instruction *NewInst = Inst.clone();
+            if (!NewInst) {
+              llvm::errs() << "Error: Failed to clone instruction.\\n";
```
Comment (aobolensk) [inline llvm/labs/lab2/isaev_dmitriy/inline_func.cpp]:
```suggestion
              llvm::errs() << "Error: Failed to clone instruction.\n";
```

File: llvm/test/lab2/isaev_dmitriy/test.ll [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  %1 = load float, ptr %a, align 4
+  %add = fadd float %1, 1.000000e+00
+  store float %add, ptr %a, align 4
+  br label %if.end
+
+if.end:                                           ; preds = %if.then, %entry
+  ret void
+}
+
+define dso_local void @_Z4bar4v() #0 {
```
Comment (aobolensk) [inline llvm/test/lab2/isaev_dmitriy/test.ll]: Add the test where you call inlined function twice (or more)

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: llvm/test/lab2/isaev_dmitriy/test.ll (hunk 2 above)
Comment (4isper) [reply llvm/test/lab2/isaev_dmitriy/test.ll]: Done

File: llvm/labs/lab2/isaev_dmitriy/inline_func.cpp [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            llvm::BasicBlock::Create(Func.getContext(), "", &Func, PostCallBB);
+        ValueMap[&CalleeBB] = CurrentBB;
+        Builder.SetInsertPoint(CurrentBB);
+        for (llvm::Instruction &Inst : CalleeBB) {
+          if (!Inst.isTerminator()) {
+            llvm::Instruction *NewInst = Inst.clone();
+            // if (!NewInst) {
+            //   llvm::errs() << "Error: Failed to clone instruction.\n";
+            //   return llvm::PreservedAnalyses::none();
+            // }
```
Comment (m-ly4) [inline llvm/labs/lab2/isaev_dmitriy/inline_func.cpp]:
```suggestion
```

File: llvm/test/lab2/isaev_dmitriy/test.ll [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+; CHECK-NEXT:   %cmp = icmp slt i32 %0, 1
+; CHECK-NEXT:   br i1 %cmp, label %if.then, label %if.end
+; CHECK: if.then:                                          ; preds = %entry
+; CHECK-NEXT:   br label %post-call
+; CHECK: 1:                                                
+; CHECK-NEXT:   %2 = alloca float, align 4
+; CHECK-NEXT:   store float 1.000000e+00, ptr %2, align 4
+; CHECK-NEXT:   %3 = load float, ptr %2, align 4
+; CHECK-NEXT:   %4 = fcmp olt float %3, 2.000000e+00
+; CHECK-NEXT:   br label %5
```
Comment (m-ly4) [inline llvm/test/lab2/isaev_dmitriy/test.ll]: This `br` instruction was conditional in non-inlined function. Please fix it.

**Review by m-ly4 — CHANGES_REQUESTED** [review]

**Review by aobolensk — APPROVED** [review]
No comments from my side. Accepted after @m-ly4 's approval

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**